### PR TITLE
Fix get tests

### PR DESCRIPTION
**Pull request by dujfalusi**
* Fixed security for `/tests` endpoints (BE and FE)
* Refactored getTests (adjusted with roles - f.e. if Admin than have highest permissions and can see everything)
* Adjusted (fixed) unit and integration tests

### CODE REVIEW

**Comment by dujfalusi** [comment]
This branch comes from `fix_sonar-qube-issues` branch because of big changes on that branch.

**Comment by stanislavgrujic** [comment]
Please update the branch with the latest develop version.

**Comment by dujfalusi** [comment]
resolve #364

File: core/src/main/java/eu/execom/testgenerator/core/repository/TestRepository.java [hunk 1]
```diff
@@ -27,4 +27,22 @@
...
+             + "(?3 IS NULL OR applicant.name LIKE ?3 OR applicant.email LIKE ?3 OR test.formula.name LIKE ?3)")
+  Page<Test> findAllActiveTestsByApplicantIdAndReviewerIdAndCompleted(Long requestedUserId, Long loggedUserId, @Nullable String query, Pageable pageRequest);
+
+  @Query("Select test from Test test join test.reviewer as reviewer where test.deleted=0 and test.completed=1 and reviewer.id=?1 and "
+             + "(?2 IS NULL OR test.applicant.name LIKE ?2 OR test.applicant.email LIKE ?2 OR test.formula.name LIKE ?2)")
+  Page<Test> findAllActiveTestsByReviewerIdAndCompleted(Long loggedUserId, @Nullable String query, Pageable pageRequest);
+
+  @Query("Select test from Test test where test.deleted=0 and test.completed=1 and "
+             + "(?1 IS NULL OR test.applicant.name LIKE ?1 OR test.applicant.email LIKE ?1 OR test.formula.name LIKE ?1)")
+  Page<Test> findAllActiveCompletedTests(@Nullable String query, Pageable pageRequest);
```
Comment (stanislavgrujic) [inline core/src/main/java/eu/execom/testgenerator/core/repository/TestRepository.java]: Please use either lower or upper case for SQL keywords, do not mix them.

File: core/src/main/java/eu/execom/testgenerator/core/repository/TestRepository.java [hunk 2]
```diff
@@ -6,25 +6,40 @@
...
-import javax.annotation.Nullable;
 
+import javax.annotation.Nullable;
 import java.util.List;
 
 public interface TestRepository extends JpaRepository<Test, Long> {
 
   @Modifying
-  @Query("UPDATE Test t set t.deleted = 1 where t.id = :id")
+  @Query("Update Test t set t.deleted = 1 where t.id = ?1")
```
Comment (stanislavgrujic) [inline core/src/main/java/eu/execom/testgenerator/core/repository/TestRepository.java]: Please make the keywords all lower case (no need to capitalize the first letter)

**Pull request merged**